pkg/deploy/generator/templates.go

```diff
@@ -0,0 +1,221 @@
+package generator
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	mgmtkeyvault "github.com/Azure/azure-sdk-for-go/services/keyvault/mgmt/2016-10-01/keyvault"
+
+	"github.com/Azure/ARO-RP/pkg/util/arm"
+)
+
+func (g *generator) _managedIdentityTemplate() *arm.Template {
+	t := templateStanza()
+
+	t.Resources = append(t.Resources,
+		g.managedIdentity(),
+	)
+
+	t.Outputs = map[string]*arm.Output{
+		"rpServicePrincipalId": {
+			Type:  "string",
+			Value: "[reference(resourceId('Microsoft.ManagedIdentity/userAssignedIdentities', 'rp-identity'), '2018-11-30').principalId]",
+		},
+	}
+	return t
+}
+
+func (g *generator) _rpTemplate() *arm.Template {
+	t := templateStanza()
+
+	params := []string{
+		"databaseAccountName",
+		"domainName",
+		"fpServicePrincipalId",
+		"rpServicePrincipalId",
+	}
+	if g.production {
+		params = append(params,
+			"extraCosmosDBIPs",
+			"extraKeyvaultAccessPolicies",
+			"mdmFrontendUrl",
+			"mdsdConfigVersion",
+			"mdsdEnvironment",
+			"pullSecret",
+			"rpImage",
+			"rpImageAuth",
+			"rpMode",
+			"sshPublicKey",
+			"vmssName",
+		)
+	} else {
+		params = append(params,
+			"adminObjectId",
+		)
+	}
+
+	for _, param := range params {
+		p := &arm.TemplateParameter{Type: "string"}
+		switch param {
+		case "extraCosmosDBIPs", "rpMode":
+			p.DefaultValue = ""
+		case "pullSecret", "rpImageAuth":
+			p.Type = "securestring"
+		}
+		t.Parameters[param] = p
+	}
+
+	if g.production {
+		t.Resources = append(t.Resources, g.pip(), g.lb(), g.vmss())
+	}
+
+	t.Resources = append(t.Resources, g.zone(),
+		g.rpvnet(), g.pevnet(),
+		g.halfPeering("rp-vnet", "rp-pe-vnet-001"),
+		g.halfPeering("rp-pe-vnet-001", "rp-vnet"))
+	if g.production {
+		t.Resources = append(t.Resources, g.cosmosdb("'ARO'")...)
+	} else {
+		t.Resources = append(t.Resources, g.cosmosdb("pparameters('databaseName')")...)
+	}
+	t.Resources = append(t.Resources, g.rbac()...)
+
+	return t
+}
+
+func (g *generator) _databaseTemplate() *arm.Template {
+	t := templateStanza()
+
+	t.Resources = append(t.Resources,
+		g.database("parameters('databaseName')", false)...)
+
+	t.Parameters = map[string]*arm.TemplateParameter{
+		"databaseAccountName": {
+			Type: "string",
+		},
+		"databaseName": {
+			Type: "string",
+		},
+	}
+
+	return t
+}
+
+func (g *generator) _rpParametersTemplate() *arm.Parameters {
```

Comment:
this isn't returning a template - call it something like `rpParameters` ?  (please do the same for the others)